okhttp-tests/src/test/java/okhttp3/internal/http2/HttpOverHttp2Test.java

```diff
@@ -1500,4 +1502,39 @@ private Buffer gzip(String bytes) throws IOException {
       }
     }
   }
+
+  /** https://github.com/square/okhttp/issues/4875 */
```

Comment:
@amirlivneh FYI, this is your commit on my PR

Reply:
Also – I got the test to pass 100 times in a row, giving me confidence I got the fix right.

Reply:
Yes, it does look promising. When I run the test locally with the fix, the test now times out one out of ~100 runs because `Callback.onFailure()` is called once and not twice as expected. When I remove the test timeout, the test sometimes hangs forever, which means one of the calls never fails and never times out. This looks like an unrelated yet real issue.